File: app/src/main/java/com/example/agiles_3/ReceipeViewModel.java
package com.example.agiles_3;

import android.app.Application;

import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;

import java.util.List;

public class ReceipeViewModel extends AndroidViewModel {
    private ReceipeRepository receipeRepository;

    private final LiveData<List<Receipe>> allReceipes;

    public ReceipeViewModel (Application application){
        super(application);
        receipeRepository = new ReceipeRepository(application);
        allReceipes = receipeRepository.getAllReceipes();
    }

    LiveData<List<Receipe>> getAllReceipes( ){
        return allReceipes;
    }

    public void insert(Receipe receipe){
        receipeRepository.insert(receipe);
    }
}

File: app/src/main/java/com/example/agiles_3/ReceipeViewModel.kt
package com.example.agiles_3

import androidx.lifecycle.*
import kotlinx.coroutines.launch

class ReceipeViewModel(private val repository: ReceipeRepository):ViewModel() {
    val allReceipes: LiveData<List<Receipe>> = repository.allRecipes.asLiveData()

    fun insert(receipe: Receipe) = viewModelScope.launch{
        //repository.insert(receipe)
    }

}

//class ReceipeViewModelFactory(private val repository: ReceipeRepository) : ViewModelProvider.Factory{
//    override fun <T : ViewModel> create(modelClass: Class<T>) :T{
//        if(modelClass.isAssignableFrom(ReceipeViewModel::class.java)){
//            @Suppress("UNCHECKED_CAST")
//            return ReceipeViewModel(repository) as T
//        }
//        throw IllegalArgumentException("Casteo del ViewModel desconocido")

//    }
//}